Doc/library/datetime.rst

```diff
@@ -765,7 +765,7 @@ Instance methods:
 .. method:: date.__format__(format)
 
    Same as :meth:`.date.strftime`. This makes it possible to specify a format
-   string for a :class:`.date` object in :ref:`formatted string
+   string for a :class:`date` object in :ref:`formatted string
```

Comment:
@erlend-aasland @hugovk I've just started working on the issue of these inconsistent links, I'm not sure what's the difference between `:class:.date` and `:class:date`. Except for this single case, the `date` class is the only one that does not use the extra dot, the others classes tend to favor the extra dot, so I'm not sure if they should or shouldn't have the dot.

Reply:
This is a feature in Sphinx where `.` means "try to find the name in the local scope first before trying the global one". See: https://www.sphinx-doc.org/en/master/usage/domains/python.html#target-resolution.

Reply:
Thanks @uatach. I vaguely remember that removing the dot prefix has been slightly controversial (I know Serhiy has some opinions here). Consider to leave them untouched.

Reply:
@erlend-aasland I understand and just to make sure there's no confusion: there are 6 types in the `datetime` module, `:class:.time` and `:class:.datetime` need the dot prefix, `:class:timedelta`, `:class:tzinfo` and `:class:timezone` do not need the dot prefix, they don't have it and I did not remove it from them, finally `:class:date` does not need the prefix, I've removed the prefix from just 2 occurrences and there are 30 other occurrences that do not have the prefix... so, regarding the controversial nature, should all occurrences (except those related to multiple links in the same paragraph) from all 6 types have the dot prefix ?

Reply:
If you need to introduce new links, leave the dot out, if it is not needed. For existing references that do not need to be touched, just leave the dot, even if it is not needed. We prefer small diffs :)

Reply:
@erlend-aasland thanks! then I think that the PR can be reviewed.